.github/workflow/deploy.yml

```
@@ -0,0 +1,199 @@
+      env:
+        GH_TOKEN: ${{ github.token }}
+      run: |
+        gh run list -c ${{ github.sha }} -w build-image --json conclusion | jq 'any(.conclusion == "success")' | grep true \
+        || (echo "::error title=The necessary build has failed::The build associated with the commit ${{ github.sha }} did not succeed. You cannot deploy this work." && exit 1)
+
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1-node16
+      with:
+        aws-region: ${{ env.AWS_REGION }}
+
+    - name: Login to Amazon ECR
+      id: login-ecr
+      uses: aws-actions/amazon-ecr-login@v1
+
+    - name: Configure Deployment Environment
+      id: config
+      uses: ./.github/actions/configure
+      with:
+        ecr-registry: ${{ steps.login-ecr.outputs.registry }}
+
+    - name: Upload Airbrake source maps
+      if: ${{ ! inputs.maintenance }}
+      env:
+        AIRBRAKE_JS_PROJECT_ID: ${{ secrets.AIRBRAKE_JS_PROJECT_ID }}
+        AIRBRAKE_JS_PROJECT_KEY: ${{ secrets.AIRBRAKE_JS_PROJECT_KEY }}
+        APPLICATION_ENV: ${{ env.application-env }}
+        TAG: ${{ env.tagged-image }}
+      run: |
+        docker run --rm \
+          --env AIRBRAKE_JS_PROJECT_ID \
+          --env AIRBRAKE_JS_PROJECT_KEY \
+          --env APPLICATION_ENV \
+          "$TAG" \
+          /site/tools/uploadSourceMaps
+
+    - name: Find last successful deployment
+      if: ${{ ! inputs.maintenance }}
+      id: prev
+      env:
+        BRANCH: ${{ env.branch }}
+      run: |
+        PREV_SHA=$(curl -X GET \
+          -H "Authorization: token ${{ github.token }}" \
+          "${GITHUB_API_URL}/repos/${{ github.repository }}/actions/workflows/${{ github.workflow }}.yml/runs?branch=${BRANCH}&status=success" \
+          | jq -r '.workflow_runs[0].head_sha')
+
+        echo "Found commit '${PREV_SHA}'."
+        echo "prev-sha=${PREV_SHA}" >> $GITHUB_ENV
+
+    - name: Put the site into maintenance mode
+      env:
+        APPLICATION_ENV: ${{ env.application-env }}
+        AWS_ACCOUNT_ID: ${{ secrets.OLD_AWS_ACCOUNT_ID }}
+        BRANCH: ${{ env.branch }}
+        PREV_SHA: ${{ env.prev-sha }}
+        TAG: ${{ env.tagged-image }}
+      run: |
+        source .github/bin/deploy.sh
+
+        MIGRATION_LOG=$(git log --no-merges --pretty=format:"- %s -- %an" \
+          "^${PREV_SHA}" HEAD -- data/migrations \
+          || echo "[Error retrieving log]")
+
+        # Put the site into maintenance mode if there are pending migrations.
+        # If there is any uncertainty, then fail safe by showing the
+        # maintenance page.
+        if [[ ! -z "${MIGRATION_LOG}" ]] || [ "${{ inputs.maintenance }}" = "true" ]; then
+          MAINTENANCE=1
+          register_task_definition
+          create_or_update_service
+        fi
```

```suggestion
        if [ "${{ inputs.maintenance }}" = "true" ]; then
          MAINTENANCE=1
          register_task_definition
          create_or_update_service
        fi
```

we don't have migrations yet